spec/s3_spec.rb

```diff
@@ -44,4 +44,91 @@
       it { expect{S3.ensure_bucket('bucket')}.to raise_error Aws::S3::Errors::Forbidden}
     end
   end
+
+  describe '#upload' do
+    it 'throws an error without a filename' do
+      expect {
+        S3.upload('bucket')
+      }.to raise_error "Must provide filename to S3.upload()"
+    end
+
+    it 'returns true with a empty file' do
+      expect(client).to receive(:put_object)
+        .with(
+          bucket: 'bucket',
+          key: 'foo',
+          body: '',
+          acl: 'private',
+        ).and_call_original
+
+      expect(
+        S3.upload('bucket', filename: 'foo')
+      ).to be true
+    end
+
+    it 'returns true with a file that has content' do
+      expect(client).to receive(:put_object)
+        .with(
+          bucket: 'bucket',
+          key: 'foo',
+          body: 'abcd',
+          acl: 'private',
+        ).and_call_original
+
+      expect(
+        S3.upload('bucket', filename: 'foo', contents: 'abcd')
+      ).to be true
+    end
+
+    it 'returns true with a file that has content and an ACL' do
+      expect(client).to receive(:put_object)
+        .with(
+          bucket: 'bucket',
+          key: 'foo',
+          body: 'abcd',
+          acl: 'public-read',
+        ).and_call_original
+
+      expect(
+        S3.upload('bucket', filename: 'foo', contents: 'abcd', acl: 'public-read')
+      ).to be true
```

Comment:
Yeah, I'm not sure either. However, returning true keeps us open for new options in the future because anything we could change to would be true. :)